{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cosmic = pd.read_csv('../data/cosmic/CosmicGenomeScreensMutantExport.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I am interested in the frequency of genetic alterations for each gene and cancer type. I am only interested in missense and nonsense mutations that are changing the protein structure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Accession Number</th>\n",
       "      <th>Gene CDS length</th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>ID_sample</th>\n",
       "      <th>ID_tumour</th>\n",
       "      <th>Primary site</th>\n",
       "      <th>Site subtype 1</th>\n",
       "      <th>Site subtype 2</th>\n",
       "      <th>...</th>\n",
       "      <th>Mutation strand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>FATHMM prediction</th>\n",
       "      <th>FATHMM score</th>\n",
       "      <th>Mutation somatic status</th>\n",
       "      <th>Pubmed_PMID</th>\n",
       "      <th>ID_STUDY</th>\n",
       "      <th>Sample Type</th>\n",
       "      <th>Tumour origin</th>\n",
       "      <th>Age</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAPSL_ENST00000397366</td>\n",
       "      <td>ENST00000397366.5</td>\n",
       "      <td>969</td>\n",
       "      <td>28375.0</td>\n",
       "      <td>HX5T</td>\n",
       "      <td>1473092</td>\n",
       "      <td>1396791</td>\n",
       "      <td>liver</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.05165</td>\n",
       "      <td>Variant of unknown origin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>323.0</td>\n",
       "      <td>surgery fresh/frozen</td>\n",
       "      <td>NS</td>\n",
       "      <td>71.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RHAG</td>\n",
       "      <td>ENST00000371175.9</td>\n",
       "      <td>1957</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>PCSI_0024_Pa_X</td>\n",
       "      <td>2121266</td>\n",
       "      <td>1995502</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.08034</td>\n",
       "      <td>Confirmed somatic variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>382.0</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RHAG</td>\n",
       "      <td>ENST00000371175.9</td>\n",
       "      <td>1957</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>S00829_1</td>\n",
       "      <td>2480818</td>\n",
       "      <td>2343563</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.95406</td>\n",
       "      <td>Confirmed somatic variant</td>\n",
       "      <td>26168399.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>surgery fresh/frozen</td>\n",
       "      <td>primary</td>\n",
       "      <td>63.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RHAG</td>\n",
       "      <td>ENST00000371175.9</td>\n",
       "      <td>1957</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>CSCC-27-T</td>\n",
       "      <td>2292469</td>\n",
       "      <td>2158533</td>\n",
       "      <td>skin</td>\n",
       "      <td>head_neck</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.07424</td>\n",
       "      <td>Confirmed somatic variant</td>\n",
       "      <td>25303977.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>surgery fresh/frozen</td>\n",
       "      <td>primary</td>\n",
       "      <td>72.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PORCN_ENST00000359882</td>\n",
       "      <td>ENST00000359882.8</td>\n",
       "      <td>1877</td>\n",
       "      <td>17652.0</td>\n",
       "      <td>T213</td>\n",
       "      <td>2658379</td>\n",
       "      <td>2518538</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.84330</td>\n",
       "      <td>Confirmed somatic variant</td>\n",
       "      <td>27149842.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>61.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene name   Accession Number  Gene CDS length  HGNC ID  \\\n",
       "0  CAPSL_ENST00000397366  ENST00000397366.5              969  28375.0   \n",
       "1                   RHAG  ENST00000371175.9             1957  10006.0   \n",
       "2                   RHAG  ENST00000371175.9             1957  10006.0   \n",
       "3                   RHAG  ENST00000371175.9             1957  10006.0   \n",
       "4  PORCN_ENST00000359882  ENST00000359882.8             1877  17652.0   \n",
       "\n",
       "      Sample name  ID_sample  ID_tumour     Primary site Site subtype 1  \\\n",
       "0            HX5T    1473092    1396791            liver             NS   \n",
       "1  PCSI_0024_Pa_X    2121266    1995502         pancreas             NS   \n",
       "2        S00829_1    2480818    2343563             lung             NS   \n",
       "3       CSCC-27-T    2292469    2158533             skin      head_neck   \n",
       "4            T213    2658379    2518538  large_intestine             NS   \n",
       "\n",
       "  Site subtype 2  ... Mutation strand SNP FATHMM prediction FATHMM score  \\\n",
       "0             NS  ...               +   n           NEUTRAL      0.05165   \n",
       "1             NS  ...               +   n           NEUTRAL      0.08034   \n",
       "2             NS  ...               +   n        PATHOGENIC      0.95406   \n",
       "3             NS  ...               +   n           NEUTRAL      0.07424   \n",
       "4             NS  ...               +   n        PATHOGENIC      0.84330   \n",
       "\n",
       "     Mutation somatic status Pubmed_PMID  ID_STUDY           Sample Type  \\\n",
       "0  Variant of unknown origin         NaN     323.0  surgery fresh/frozen   \n",
       "1  Confirmed somatic variant         NaN     382.0                    NS   \n",
       "2  Confirmed somatic variant  26168399.0       NaN  surgery fresh/frozen   \n",
       "3  Confirmed somatic variant  25303977.0       NaN  surgery fresh/frozen   \n",
       "4  Confirmed somatic variant  27149842.0       NaN                    NS   \n",
       "\n",
       "  Tumour origin    Age  \n",
       "0            NS  71.00  \n",
       "1            NS    NaN  \n",
       "2       primary  63.00  \n",
       "3       primary  72.00  \n",
       "4            NS  61.25  \n",
       "\n",
       "[5 rows x 36 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cosmic.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
